store/src/db/sql.rs

```diff
@@ -125,16 +155,83 @@ pub fn select_accounts_by_block_range(
 /// transaction.
 pub fn upsert_accounts(
     transaction: &Transaction,
-    accounts: &[(AccountId, RpoDigest)],
+    accounts: &[(AccountId, Option<AccountDetails>, RpoDigest)],
     block_num: BlockNumber,
 ) -> Result<usize> {
-    let mut stmt = transaction.prepare("INSERT OR REPLACE INTO accounts (account_id, account_hash, block_num) VALUES (?1, ?2, ?3);")?;
+    let mut acc_stmt = transaction.prepare(
+        "INSERT OR REPLACE INTO accounts (account_id, account_hash, block_num) VALUES (?1, ?2, ?3);",
+    )?;
+    let mut select_details_stmt = transaction.prepare(
+        "SELECT nonce, vault, storage, code FROM account_details WHERE account_id = ?1;",
+    )?;
+    let mut new_details_stmt = transaction.prepare(
+        "INSERT INTO account_details (account_id, nonce, vault, storage, code) VALUES (?1, ?2, ?3, ?4, ?5);"
+    )?;
+    let mut update_details_stmt = transaction.prepare(
+        "UPDATE account_details SET nonce = ?2, vault = ?3, storage = ?4 WHERE account_id = ?1;",
+    )?;
 
     let mut count = 0;
-    for (account_id, account_hash) in accounts.iter() {
-        count +=
-            stmt.execute(params![u64_to_value(*account_id), account_hash.to_bytes(), block_num])?
+    for (account_id, details, account_hash) in accounts.iter() {
+        let account_id = *account_id;
+        count += acc_stmt.execute(params![
+            u64_to_value(account_id),
+            account_hash.to_bytes(),
+            block_num
+        ])?;
+        if let Some(details) = details {
+            match details {
+                AccountDetails::Full(account) => {
```

Comment:
nit: this could be just:

```rust
match details {
  Some(AccountDetails::Full(account)) => { ... },
  Some(AccountDetails::Delta(delta)) => { ... },
  None => (),
}
```

Not a huge improvement, but this makes the indentation shallower which IMO is a bit easier to read.